File: robocode2-bot-api-java/src/main/java/net/robocode2/events/BulletFiredEvent.java
package net.robocode2.events;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Value;
import net.robocode2.BulletState;

/** Event occurring when a bullet has been fired from a bot */
@Value
@EqualsAndHashCode(callSuper = true)
public class BulletFiredEvent extends Event {
  /** Bullet that was fired */
  BulletState bullet;

  @Builder
  private BulletFiredEvent(int turnNumber, BulletState bullet) {
    this.turnNumber = turnNumber;
    this.bullet = bullet;
  }
}

File: robocode-tankroyale-bot-api-csharp/events/BulletFiredEvent.cs
namespace Robocode.TankRoyale.BotApi
{
  /// <summary>
  /// Event occurring when a bullet has been fired from a bot.
  /// </summary>
  public sealed class BulletFiredEvent : Event
  {
    /// <summary>Bullet that was fired.<summary>
    public BulletState Bullet { get; }

    /// <summary>
    /// Constructor.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="bullet">Bullet that was fired.</param>
    public BulletFiredEvent(int turnNumber, BulletState bullet) : base(turnNumber) => Bullet = bullet;
  }
}